sdk/core/azure-core-tracing-opentelemetry/tests/test_tracing_implementations.py

```diff
@@ -0,0 +1,161 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""The tests for opencensus_span.py"""
+
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+from opentelemetry.trace import SpanKind as OpenTelemetrySpanKind
+
+from azure.core.tracing.ext.opentelemetry_span import OpenTelemetrySpan
+from azure.core.tracing import SpanKind
+import os
+
+import pytest
+
+
+class TestOpentelemetryWrapper:
+    def test_span_passed_in(self, tracer):
+        with tracer.start_as_current_span(name="parent") as parent:
+            wrapped_span = OpenTelemetrySpan(parent)
+
+            assert wrapped_span.span_instance.name == "parent"
+            assert parent is tracer.get_current_span()
+            assert wrapped_span.span_instance is tracer.get_current_span()
+
+            assert parent is tracer.get_current_span()
+
+    def test_no_span_passed_in_with_no_environ(self, tracer):
+        with tracer.start_as_current_span("Root") as parent:
+            with OpenTelemetrySpan() as wrapped_span:
+
+                assert wrapped_span.span_instance.name == "span"
+                assert wrapped_span.span_instance is tracer.get_current_span()
+
+            assert parent is tracer.get_current_span()
+
+
+    def test_span(self, tracer):
+        with tracer.start_as_current_span("Root") as parent:
+            assert OpenTelemetrySpan.get_current_tracer() is tracer
+            with OpenTelemetrySpan() as wrapped_span:
+                assert wrapped_span.span_instance is tracer.get_current_span()
+
+                with wrapped_span.span() as child:
+                    assert child.span_instance.name == "span"
+                    assert child.span_instance is tracer.get_current_span()
+                    assert child.span_instance.parent is wrapped_span.span_instance
+
+    def test_start_finish(self, tracer):
+        with tracer.start_as_current_span("Root") as parent:
+            wrapped_class = OpenTelemetrySpan()
+            assert wrapped_class.span_instance.end_time is None
```

Comment:
is `start_time` value explicitly undefined before `start()` is called?

Reply:
start_time is set at instance creation line 57. I added more assert for clarity